src/Inciter/Scheme.h

```diff
@@ -231,9 +231,9 @@ class Scheme : public SchemeBase {
     //!   specifying a non-default last argument.
     template< typename Op, typename... Args, typename std::enable_if<
       std::is_same< Op, tag::elem >::value, int >::type = 1,
-      typename std::enable_if< sizeof...(Args) == 11, int >::type = 0 >
+      typename std::enable_if< sizeof...(Args) == 12, int >::type = 0 >
```

Comment:
Changing these doesnt seem like much fun!

Reply:
Indeed. This is something I'm not very proud of with the work that went into the `Scheme`-stuff: everything is generic, except for the number of arguments passed.

Reply:
Now that I think about it more, this could possibly be fixed in some clever way. That `enable_if` is really only there to enable the compiler differentiate between the overloads of which one does not take a pointer at the end of the entry method argument list and one that does. Scheme provides both so client code is simpler. But now I'm thinking for better maintainability and readability, may be we should just require the client code to pass that final pointer always (which of course can be `nullptr`). What do you think?

Reply:
Indeed. This is something I'm not very proud of with the work that went into the Scheme-stuff: everything is generic, except for the number of arguments passed.

Now that I think about it more, this could possibly be fixed in some clever way. That enable_if is really only there to enable the compiler differentiate between the overloads of which one does not take a pointer at the end of the entry method argument list and one that does. `Scheme` provides both so client code is simpler. But now I'm thinking for better maintainability and readability, may be we should just require the client code to pass that final pointer always (which of course can be `nullptr`). What do you think?

Reply:
I'm not sure I fully understand this. Do you mean we have two methods, one which takes a pointer, and one which does not. We should change the interface to always pass a pointer, which can be nullptr in the instance we don't actually need it? I'm OK with that, but don't immediately see that it's definitely "nicer"

Reply:
Yeah, we have two overloads for a single Charm++ entry method: one takes an extra optional argument which is a pointer to an options-struct and one for which these extra options are not passed, so Charm++ just uses the default. I wanted to mirror the Charm++ API with it. But if I remove the overload (and leave the one with the pointer to allow for the options), I believe the `enable_if` can go away. In that case client code will always have to pass a pointer, which can be `nullptr`.

Example client code:
```
m_scheme.discInsert< tag::elem >( ..., nullptr );  // passing the extra options-pointer as nullptr (this is how it wil have to be if we remove one overload, which removes the need for enable_if)
m_scheme.discInsert< tag::elem >( ... );    // not passing the extra options-pointer (works now)
```